src/modules/mavlink/mavlink_messages.cpp

```diff
@@ -5310,6 +5311,179 @@ class MavlinkStreamRawRpm : public MavlinkStream
 	}
 };
 
+class MavlinkStreamESCInfo : public MavlinkStream
+{
+public:
+	const char *get_name() const override
+	{
+		return MavlinkStreamESCInfo::get_name_static();
+	}
+
+	static constexpr const char *get_name_static()
+	{
+		return "ESC_INFO";
+	}
+
+	static constexpr uint16_t get_id_static()
+	{
+		return MAVLINK_MSG_ID_ESC_INFO;
+	}
+
+	uint16_t get_id() override
+	{
+		return get_id_static();
+	}
+
+	static MavlinkStream *new_instance(Mavlink *mavlink)
+	{
+		return new MavlinkStreamESCInfo(mavlink);
+	}
+
+	unsigned get_size() override
+	{
+		static constexpr unsigned size_per_batch = MAVLINK_MSG_ID_ESC_INFO_LEN + MAVLINK_NUM_NON_PAYLOAD_BYTES;
+		return _esc_status_sub.advertised() ? size_per_batch * number_of_batches : 0;
+	}
+
+private:
+	uORB::Subscription _esc_status_sub{ORB_ID(esc_status)};
+
+	/* do not allow top copying this class */
+	MavlinkStreamESCInfo(MavlinkStreamESCInfo &) = delete;
+	MavlinkStreamESCInfo &operator = (const MavlinkStreamESCInfo &) = delete;
+
+protected:
+	uint8_t number_of_batches{0};
```

Comment:
```suggestion
	uint8_t _number_of_batches{0};
```

Reply:
Had my doubts on that one. I will correct ;)